File: PVPC-Bungee-Tools/src/main/java/pl/pvpcloud/tools/listeners/BlazingPackHandshakeAuthListener.java
package pl.pvpcloud.tools.listeners;

import net.md_5.bungee.api.chat.BaseComponent;
import net.md_5.bungee.api.plugin.Listener;
import net.md_5.bungee.event.EventHandler;
import pl.blazingpack.bpauth.BlazingPackHandshakeAuthEvent;
import pl.pvpcloud.tools.ToolsPlugin;

public class BlazingPackHandshakeAuthListener implements Listener {

    private final ToolsPlugin plugin;

    public BlazingPackHandshakeAuthListener(ToolsPlugin plugin) {
        this.plugin = plugin;
    }

    @EventHandler
    public void onBlazingPackHandshakeAuth(BlazingPackHandshakeAuthEvent event) {
        if (!event.isUsingBlazingPack() && this.plugin.config.getWhiteListPlayers().contains(event.getInitialHandler().getName().toLowerCase())) {
            BaseComponent[] no_msg = null;
            event.setCancelReason(no_msg);
        }
    }
}

File: PVPC-Bungee-Auth/src/main/kotlin/pl/pvpcloud/auth/listeners/BlazingPackHandshakeAuthListener.kt
package pl.pvpcloud.auth.listeners

import net.md_5.bungee.api.plugin.Listener
import net.md_5.bungee.event.EventHandler
import pl.blazingpack.bpauth.BlazingPackHandshakeAuthEvent
import pl.pvpcloud.auth.AuthPlugin
import pl.pvpcloud.auth.helpers.PremiumHelper
import pl.pvpcloud.auth.holders.PlayerPremiumHolder

class BlazingPackHandshakeAuthListener(private val plugin: AuthPlugin) : Listener {

    @EventHandler
    fun onBlazingPackHandshakeAuth(event: BlazingPackHandshakeAuthEvent) {
        if (event.isUsingBlazingPack) {
            PlayerPremiumHolder.players.put(event.initialHandler.name, event.isPremium)
            event.initialHandler.isOnlineMode = event.isPremium
        } else {
            val hasPaid = false
            PlayerPremiumHolder.players.put(event.initialHandler.name, hasPaid)
            event.initialHandler.isOnlineMode = hasPaid
        }
    }
}